controllers/authController.js

```diff
@@ -0,0 +1,79 @@
+const jwt = require("jsonwebtoken");
+const userTemplateCopy = require("../schema/userSchema");
+const bcrypt = require("bcrypt");
+const { isInValid, addUserService } = require("../services/userServices");
+const {
+  generateTokens,
+  generateAccessToken,
+} = require("../services/authServices");
+
+const login = async (req, res) => {
+  const { emp_code, password } = req.body;
+
+  if (!(emp_code && password))
+    return res.status(400).json({ msg: "Bad Request" });
+
+  const dbUser = await userTemplateCopy.find({ emp_code });
+
+  if (!dbUser) return res.status(401).json({ msg: "Invalid Employee Code" });
+
+  let newObj = { ...dbUser }[0];
+
+  bcrypt.compare(newObj.password, password, (err, result) => {
+    if (err) {
+      res.status(500).json({ msg: "Error comparing passwords:", err });
+    } else {
+      if (!result) {
+        res.status(401).json({ msg: "Invalid Password", err });
+      }
+    }
+  });
+
+  res.json({
+    msg: "Welcome " + newObj.name,
+    tokens: await generateTokens(req.body),
+  });
+};
+
+const signup = async (req, res) => {
+  const userData = req.body;
+  if (isInValid(userData)) {
+    res.status(400).json({ msg: "Bad request" });
+    return;
+  }
+  try {
+    // Check Duplication of Employee Code
+    const dbUser = await userTemplateCopy.find({ emp_code: userData.emp_code });
```

Comment:
Same, can we use findOne here?

Reply:
Yah, Ram that will be very much suitable here.